CONTRIBUTING.md

```diff
@@ -13,31 +15,52 @@ Operator Lifecycle Manager is Apache 2.0 licensed and accepts contributions via
 
-Reporting bugs is one of the best ways to contribute. However, a good bug report has some very specific qualities, so please read over our short document on [reporting bugs](./doc/dev/reporting_bugs.md) before submitting a bug report. This document might contain links to known issues, another good reason to take a look there before reporting a bug.
+Reporting bugs is one of the best ways to contribute. However, a good bug report has some very specific qualities, so please read over our short document on [reporting bugs](./doc/dev/reporting_bugs.md) before submitting a bug report. Before filing a bug report, ensure the bug hasn't already been reported by searching through the operator-lifecycle-manager project [Issues][issues].
+
+Any new contribution should be accompanied by a new or existing issue. This issue can help track work, discuss the design and implementation, and help avoid wasted efforts or multiple people working on the same issue, compared to submitting a PR first. Trivial changes, like fixing a typo in the documentation, do not require the creation of a new issue. Proposing larger changes to the operator-lifecycle-manager project may require an enhancement be created in the [operator-framework/enhancements](https://github.com/operator-framework/enhancements/) repository.
 
 ## Contribution flow
 
 This is a rough outline of what a contributor's workflow looks like:
 
-- Create a topic branch from where to base the contribution. This is usually master.
+- Identify or create an issue.
+- Create a topic branch from where to base the contribution. This is usually the master branch.
 - Make commits of logical units.
 - Make sure commit messages are in the proper format (see below).
+- Ensure all relevant commit messages contain a valid sign-off message (see below).
 - Push changes in a topic branch to a personal fork of the repository.
-- Submit a pull request to operator-framework/operator-lifecycle-manager.
-- The PR must receive a LGTM from two maintainers found in the MAINTAINERS file.
+- Submit a pull request to the operator-framework/operator-lifecycle-manager repository.
+- Wait and respond to feedback from the maintainers listed in the OWNERS file.
 
 Thanks for contributing!
 
+### Code Review
+
+Each PR must be labeled with at least one "lgtm" label and at least one "approved" label before it can be merged.
+
+Maintainers that have approval permissions are listed in the "approvers" column in the root [OWNERS][owners] file.
+
 ### Code style
 
-The coding style suggested by the Golang community is used in operator-lifecycle-manager. See the [style doc](https://github.com/golang/go/wiki/CodeReviewComments) for details.
+The coding style suggested by the Golang community is used in the operator-lifecycle-manager project. See the [style doc](https://github.com/golang/go/wiki/CodeReviewComments) for details.
+
+Please follow this style to make the operator-lifecycle-manager project easier to review, maintain and develop.
```

Comment:
Should we add a reference to `go-fmt` here?

Reply:
If we're running down that route, it's likely best to call out what checks (e.g. Makefile targets) should be run before submitting a code-changing PR? Mainly thinking along the lines that verify the sanity-type checks that slow down review process should be done before submission.

Reply:
If we have a `fmt` make target, we can abstract the tool choice here.

Related: eventually, we should gate PRs on lint status.

Reply:
Ah, weirdly enough it looks like we _don't_ have a check enforcing `go fmt` linting checks. I can update that verification check to also run `go fmt` (and maybe down the line, `go imports`?) outside of this PR. I'll still call out that's largely what we base style for Go packages these days.

Reply:
I didn't see Nick's reply in time when I was writing out mine - completely agree, we should be gating PRs on any linting/sanity/etc. type checks, like we currently do in the registry codebase.

Reply:
I'm going to add a TODO comment for now - I'll update this document once I get to adding that linting check.

Reply:
Once https://github.com/operator-framework/operator-lifecycle-manager/pull/2346 merges I'll update this document for what kind of checks should be run locally before submitting a PR (or introducing/changes/etc. new commits) so review time is shortened.